File: values.py
def values_for_key(list_of_dicts, key):

    # output = []

    # for i in list_of_dicts:
    #     if key in i:
    #         output.append(i[key])
    #     else:
    #         output.append(None)

    # return output

    return [i[key] if key in i else None for i in list_of_dicts]

dicts = [{"age": 10}, {"remote": False}]
# dicts = [{"age": 10}, {"age": 12}]

print(values_for_key(dicts, "age"))

File: values.js
function valuesForProperty(values, propertyName) {

    // let output = [];
    // // console.log(values.length);

    // for (let i of values) {
    //     if (i.hasOwnProperty(propertyName)) {
    //         output.push(i[propertyName]);
    //     } else {
    //         output.push(undefined);
    //     }
    // }
    // return output;


    return values.map(value => {
        if (Object.keys(value) == propertyName)
        {
            return Object.values(value)[0];
        } else {
            return undefined;
        }

    });

}

let dicts = [{age: 10}, {remote: false}]
// let dicts = [{"age": 10}, {"age": 12}]
// let dicts = []

console.log(valuesForProperty(dicts, "age"))
